### PR TITLE
implement ducc stats when .fr/.kille - does not send command yet

### PR DESCRIPTION
**Pull request by aboft**
implemented duck stats for friends and kills along with components. Does not currently send messages to trigger score events. This only parses messages at this time.

### CODE REVIEW

File: backend/irc/ircconnection.ts [hunk 1]
```diff
@@ -85,6 +86,24 @@ rl.on('line', async (line) => {
...
+        myEmitter.emit('friendScore', splitMsgByBullet);
+        await DatabaseDuccUtils.retrieveAllDuccScores();
+      } else if (msg.match(/Duck killer scores in #/i)) {
+        const splitMsgByBullet: string[] = msg.split('\u2022');
+        // fix the first split by removing the 'Duck ... scores in #channel'
+        const correctFirstScore: string[] = splitMsgByBullet[0].split(':');
+        splitMsgByBullet[0] = `${correctFirstScore[2]}: ${correctFirstScore[3]}`;
+        await insertOrUpdateKillerScores(splitMsgByBullet);
+        myEmitter.emit('killedScore', splitMsgByBullet);
+      }
```
Comment (LoLei) [inline backend/irc/ircconnection.ts]: These are so similar they could be made into one function with a friend/killer argument.

File: backend/irc/utils/db/DuccScores.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        .where({ user: score[0] })
+        .then(async (rows) => {
+          if (rows.length === 0) {
+            await knex('killer_scores').insert({ user: score[0], duccs });
+          } else {
+            await knex('killer_scores').where({ user: score[0] }).update({ duccs });
+          }
+        });
+    });
+  }
```
Comment (LoLei) [inline backend/irc/utils/db/DuccScores.ts]: Could also be unified

File: backend/server.ts [hunk 3]
```diff
@@ -32,8 +32,8 @@ app.get('/test', async (req, res: Response<any, number>) => {
   await Sender.sendMessages(res);
   await Sender.sendLineCounts(res);
   await Sender.sendTopWords(res);
+  await Sender.sendDuccScores(res); // Daily
 
-  // Daily
   setInterval(async (_) => {
```
Comment (LoLei) [inline backend/server.ts]: I assume these are auto formatter changes. In this case it doesn't really make sense. Maybe move the comment next to line 37 so it's satisfied.

File: frontend/components/DuccKillerStatsList.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <div>
+      <div className={"userlist-header"}>Top 10 Ducc Killers: </div>
+      <div className={"userlist-content"}>
+        {fetchedUsers.map((user) => (
+          <DuccStats key={user.user} user={user.user} duccs={user.duccs} />
+        ))}
+      </div>
+    </div>
+  );
+}
```
Comment (LoLei) [inline frontend/components/DuccKillerStatsList.tsx]: DuccFriendStatsList.tsx and DuccKillerStatsList.tsx could be made into one component

File: frontend/components/DuccFriendStatsList.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react";
+import DuccStats, { IDucc } from "./DuccStats";
+import eventSource from "../data/EventSource";
+
+export default function DuccFriendStatsList() {
+  const [fetchedUsers, setFetchedUsers] = useState<IDucc[]>([
```
Comment (LoLei) [inline frontend/components/DuccFriendStatsList.tsx]: I wouldn't call these "Users" here. Maybe "Scores"?

**Review by LoLei — CHANGES_REQUESTED** [review]

File: frontend/components/DuccKillerStatsList.tsx (hunk 4 above)
Comment (aboft) [reply frontend/components/DuccKillerStatsList.tsx]: I was pulling my hair out when trying to figure it out even though it should be something simple. Would you mind giving it a try?

File: ijdq:q [hunk 6]
```diff
@@ -0,0 +1,4 @@
+  delete[m
+  ducc-stats[m
+* [32mducc-stats-new[m
+  lmrdashboard[m
```
Comment (LoLei) [inline ijdq:q]: lol wtf is this

**Review by LoLei — APPROVED** [review]

**Pull request merged**